WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.java

```diff
@@ -0,0 +1,411 @@
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onWPOrgPluginFetched(PluginStore.OnWPOrgPluginFetched event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching the wporg plugin with type: " + event.error.type);
+            return;
+        }
+
+        if (!TextUtils.isEmpty(event.pluginSlug)) {
+            mLastUpdatedWpOrgPluginSlug.setValue(event.pluginSlug);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginDirectoryFetched(PluginStore.OnPluginDirectoryFetched event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while fetching the plugin directory: " + event.type);
+            if (event.type == PluginDirectoryType.NEW) {
+                mNewPluginsListStatus.setValue(PluginListStatus.ERROR);
+            } else if (event.type == PluginDirectoryType.POPULAR) {
+                mPopularPluginsListStatus.setValue(PluginListStatus.ERROR);
+            }
+            return;
+        }
+        PluginListStatus listStatus = event.canLoadMore ? PluginListStatus.CAN_LOAD_MORE : PluginListStatus.DONE;
+        switch (event.type) {
+            case NEW:
+                reloadNewPlugins();
+                mNewPluginsListStatus.setValue(listStatus);
+                break;
+            case POPULAR:
+                reloadPopularPlugins();
+                mPopularPluginsListStatus.setValue(listStatus);
+                break;
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginDirectorySearched(PluginStore.OnPluginDirectorySearched event) {
+        if (!mSearchQuery.equals(event.searchTerm)) {
+            return;
+        }
+        if (event.isError()) {
+            AppLog.e(AppLog.T.PLUGINS, "An error occurred while searching the plugin directory");
+            mSearchPluginsListStatus.setValue(PluginListStatus.ERROR);
+            return;
+        }
+        mSearchResults.setValue(event.plugins);
+        mSearchPluginsListStatus.setValue(PluginListStatus.DONE);
+    }
+
+    // Search
+
+    public void setSearchQuery(String searchQuery) {
+        mSearchQuery = searchQuery;
+
+        // Don't delay if the searchQuery is empty
+        submitSearch(searchQuery, !TextUtils.isEmpty(searchQuery));
+    }
+
+    private boolean shouldSearch() {
+        // We need at least 2 characters to be able to search plugins
+        return getSearchQuery() != null && getSearchQuery().length() > 1;
+    }
+
+    private void submitSearch(@Nullable final String query, boolean delayed) {
+        if (delayed) {
+            mHandler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    if (StringUtils.equals(query, getSearchQuery())) {
+                        submitSearch(query, false);
```

Comment:
We can make `submitSearch` protected or package-private to avoid creating synthetic accessor methods.

Reply:
All synthetic accessor method issues should be fixed in 8244bd59d148e7e560c82951daded148e249208f.